### ISSUE TITLE
`cfr_rolling()` fails when cumulative case number is 0

### ISSUE THREAD

**Issue by Bisaloo** [epiverse-trace/cfr#97]
This can arise in situation like this one:

``` r
library(cfr)

data("covid_data")
df_covid_uk <- covid_data[covid_data$country == "United Kingdom", ]

res <- cfr_rolling(
  data = df_covid_uk
)
#> Error in (function (x, n, p = 0.5, alternative = c("two.sided", "less", : 'n' must be a positive integer >= 'x'

df_covid_uk <- df_covid_uk |>
  subset(cumsum(cases) > 0)

res <- cfr_rolling(
  data = df_covid_uk
)
head(res)
#>   severity_mean severity_low severity_high
#> 1           1.0   0.02500000     1.0000000
#> 2           1.0   0.02500000     1.0000000
#> 3           1.0   0.02500000     1.0000000
#> 4           1.0   0.15811388     1.0000000
#> 5           0.1   0.01234853     0.3169827
#> 6           0.1   0.01234853     0.3169827
```

<sup>Created on 2023-10-30 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by pratikunterwegs** [comment]
Thanks for noticing this - I'm proposing to fix this using the same logic as `cfr_time_varying()`, which works on this data out of the box. The fix would only run the binomial test at indices where deaths < cases AND cases > 0.

**Comment by pratikunterwegs** [comment]
Fixed in #98
